CONTRIBUTING.md

```diff
@@ -71,18 +88,33 @@ Remember to always work in a branch of your local copy, as you might otherwise h
 
 Please also see [GitHub workflow](https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#github-workflow) section of general project contributing guide.
 
-### Running the tests
+## Development
+
+### Tools used
+
+- [NPM](https://npmjs.com)
+- [TypeScript](https://www.typescriptlang.org/)
+- [lerna](https://github.com/lerna/lerna) to manage dependencies, compilations, and links between packages. Most lerna commands should be run by calling the provided npm scripts.
+- [npm workspaces](https://docs.npmjs.com/cli/v10/using-npm/workspaces)
+- [MochaJS](https://mochajs.org/) for tests
+- [eslint](https://eslint.org/)
+
+### General guidance
 
 The `opentelemetry-js-contrib` project is written in TypeScript.
 
-- `npm install` to install dependencies.
+As a general rule, installing and then compiling from the root directory should always be done first before anything else.
+After installing and compiling from the root directory, do it again from the specific package directory you are working in.
```

Comment:
>  do it again from the specific package directory you are working in.

Do you find that is necessary after a build at the top-level? It is after making any local changes, but not otherwise in my experience so far.

Reply:
At this point I realized I've been doing it out of habit, so it's certainly possible that it's only necessary to do it after local changes. So maybe something like always install and compile from the root, and then install and compile in the local package after any changes are made